frontend/app/component/form/pollingstation_save/PollingStationSaveForm.tsx

```diff
@@ -13,29 +18,69 @@ export function PollingStationSaveForm() {
     });
   }, [registerCurrentForm]);
 
+  const finalisationAllowed = Object.values(formState.sections).every(
+    (section) =>
+      section.errors.length === 0 && (section.warnings.length === 0 || section.ignoreWarnings),
+  );
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) =>
+    void (async (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+
+      if (!finalisationAllowed) return;
+
+      await finaliseDataEntry();
+      navigate(`/${election.id}/input#data_entry_saved`);
+    })(event);
+
   return (
-    <div>
+    <Form onSubmit={handleSubmit} id="check_save_form">
       <h2>Controleren en opslaan</h2>
 
+      {/* TODO: #103 format this according to design */}
       <p>Hieronder zie je een overzicht van alle eventuele fouten en waarschuwingen.</p>
+      <ul>
+        {Object.values(formState.sections).map((section) => (
+          <li key={section.id}>
+            <span>{section.title || section.id}</span>
+            <ul>
+              <li>
+                <span>Fouten:</span>
+                <ul>
+                  {section.errors.map((error, n) => (
+                    <li key={`error${n}`}>{error.code}</li>
+                  ))}
+                </ul>
+              </li>
+              <li>
+                <span>Waarschuwingen:</span>
+                <ul>
+                  {section.warnings.map((warning, n) => (
+                    <li key={`warning${n}`}>{warning.code}</li>
+                  ))}
+                </ul>
+              </li>
+              <li>Waarschuwingen geaccepteerd: {section.ignoreWarnings ? "Ja" : "Nee"}</li>
+            </ul>
+          </li>
+        ))}
+      </ul>
+      {finalisationAllowed ? (
+        <p>Je kan de resultaten van dit stembureau opslaan.</p>
+      ) : (
+        <p>Je kan de resultaten van dit stembureau nog niet opslaan.</p>
+      )}
 
-      {Object.values(formState.sections).map((section) => (
-        <div key={section.id}>
-          <h3>{section.title || section.id}</h3>
-          <h4>Errors:</h4>
-          {section.errors.map((error, n) => (
-            <p key={`error${n}`}>{error.code}</p>
-          ))}
-          <h4>Warnings:</h4>
-          {section.warnings.map((warning, n) => (
-            <p key={`warning${n}`}>{warning.code}</p>
-          ))}
-
-          <br />
-          <br />
-          <br />
-        </div>
-      ))}
-    </div>
+      {finalisationAllowed && (
+        <BottomBar type="input-grid">
+          <BottomBar.Row>
+            <Button type="submit" size="lg" disabled={status.current === "finalising"}>
+              Opslaan
+            </Button>
+            <KeyboardKeys keys={[KeyboardKey.Shift, KeyboardKey.Enter]} />
```

Comment:
Submitting with Shift+Enter does not work for me.

Reply:
That should be provided by the `Form` component. If that component doesn't work maybe we can fix that in a separate issue, where we can also make sure to write test coverage for the keyboard shortcut?

Reply:
We do have test coverage for shift+enter for the four forms and those tests pass. So not sure what's different about this form.

Reply:
Ok, then I'll write a test for this form too and will try to solve out the issue.

Reply:
Fixed in 4e89eca221471986ddf6d25ff29c8c3033252869 by attaching the keyboard shortcut listener to the document instead of the `<form>` node. @lkleuver what do you think of that solution?

Reply:
yeah thats fine, it sort of felt safe to have the listener only active when the form is active, but this leads to unexpected behavior if you clicked outside the form and the shortcut no longer works.